[UXUI-171] Lead space components

andersonjeccel opened the pull request:
<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 7.x)
* a.b for any bug fixes (e.g. 5.2, 6.0)
* c.x for any bug fixes, features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 7.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ❌ <!-- Use emojis to indicate positive (green) or negative (red) for each item in the table. -->
| New feature/enhancement? (use the a.x branch)      | ✔️
| Deprecations?                          | ❌
| BC breaks? (use the c.x branch)        | ❌
| Automated tests included?              | ✔️ <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      | mautic/user-documentation#... <!-- required for new features -->
| Related developer documentation PR URL | mautic/developer-documentation-new#... <!-- required for developer-facing changes -->
| Issue(s) addressed                     | Fixes #... <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

## Description

This PR adds lead space, a UI component that allows developers to show banners with heading, copy and background image. My goal is to help Mautic become more visually appealing without cluttering highly used places.

I'm implementing in 3 situations for now:

#### When no identified contacts exist on the database _(I'm open to change the image upon feedback)_as part of the introduction content, disappearing after at least 1 is present
![image](https://github.com/user-attachments/assets/7f83c12c-fde4-4be1-a3b5-c92860e7ae84)

#### When at least 1 webhook is created, otherwise is not shown
![image](https://github.com/user-attachments/assets/6a9b45a6-43f7-4da6-80a1-c45a55de6f21)

#### When at least 1 API credential is created, otherwise is not shown
![image](https://github.com/user-attachments/assets/19e0169e-ba4a-4e3b-9aae-dcd7e628b0ed)

### Considerations
- Moved all introduction content to a separate Twig file
- Added images to the Assets/images folder within each bundle
- Created a Twig extension to return the contrasted color against a provided hex value

<!--
Please write a short README for your feature/bugfix. This will help people understand your PR and what it aims to do. If you are fixing a bug and if there is no linked issue already, please provide steps to reproduce the issue here.
-->
<!-- Remove HTML comment markup below to use the table for screenshots when relevant. -->
<!--
| Before                                 | After
| -------------------------------------- | ---
|                                        | 
-->


---
### 📋 Steps to test this PR:

<!--
This part is crucial. Take the time to write very clear, annotated and step by step test instructions, because testers may not be developers.
-->
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Contacts with the list empty > See the banner
3. Create a contact and go back to the list to see it disappeared
4. Open cog menu at the top tight of the screen to see admin menu options > Webhooks 
5. Click New and fill required fields to create your webhook > Save & Close
6. Go back to the webhooks page and see the banner
7. Still under the admin menu, open Configuration
8. Find the tab API settings, and enable the toggle for "API enabled?"
9. Open the admin menu again, and click API Credentials
10. Click new and fill required info > Save & Close
11. Go back to the API Crendetials list and see the banner

<!--
If you have any deprecations and backwards compatibility breaks, list them here along with the new alternative.
-->

andersonjeccel committed:
Styles

andersonjeccel committed:
Styles

andersonjeccel committed:
Add lead space template

andersonjeccel committed:
Add lead space template

andersonjeccel committed:
wip test

andersonjeccel committed:
wip test

andersonjeccel committed:
Moving to rem-based unit sizes

andersonjeccel committed:
Moving to rem-based unit sizes

andersonjeccel committed:
Styles

andersonjeccel committed:
Styles

andersonjeccel committed:
Update lead-space.html.twig

andersonjeccel committed:
Update lead-space.html.twig

andersonjeccel committed:
Optimizing the template

andersonjeccel committed:
Optimizing the template

andersonjeccel committed:
Styles

andersonjeccel committed:
Styles

andersonjeccel committed:
Changing class used for rendering full width buttons

andersonjeccel committed:
Changing class used for rendering full width buttons

andersonjeccel committed:
Implementing color aware variable for lead space

andersonjeccel committed:
Implementing color aware variable for lead space

andersonjeccel committed:
[Contacts] Migrating introduction content to separate file

andersonjeccel committed:
[Contacts] Migrating introduction content to separate file

andersonjeccel committed:
make media alt option in lead space template

andersonjeccel committed:
make media alt option in lead space template

andersonjeccel committed:
wip

andersonjeccel committed:
wip

andersonjeccel committed:
Moving introduction contents to separate files

andersonjeccel committed:
Moving introduction contents to separate files

andersonjeccel committed:
Merge branch 'lead-space-components' of https://github.com/andersonjeccel/mautic into lead-space-components

andersonjeccel committed:
Revert "Moving to rem-based unit sizes"

This reverts commit 8716e42982c2e54e11c019c0bb4b4c22ecc16157.

andersonjeccel committed:
Fixing contained list padding

andersonjeccel committed:
Moving leadspace rem to px

andersonjeccel committed:
Creating ColorsExtension to get contrast from color

andersonjeccel committed:
Applying color function to template

andersonjeccel committed:
Applying lead space to API page

andersonjeccel committed:
Applying lead space to Webhooks page

andersonjeccel committed:
Revert "wip"

andersonjeccel committed:
Finishing lead space for Contacts page

andersonjeccel committed:
Removing alt attribute from lead space for API and webhooks

andersonjeccel committed:
Merge branch '7.x' into lead-space-components

andersonjeccel committed:
phpstan, rector, cs fixer

andersonjeccel committed:
Adding test for ColorExtension

andersonjeccel committed:
rector, phpstan and cs fix for test

andersonjeccel committed:
Merge 7.x branch into lead-space-components - resolved all conflicts by choosing introduction files over content-block approach

andersonjeccel committed:
Change template path

andersonjeccel committed:
csfix

andersonjeccel committed:
Merge branch '7.x' into lead-space-components

RCheesley committed:
Merge branch '7.x' into lead-space-components

RCheesley reviewed (changes requested):
I think maybe this needs a bit of tweaking, I couldn't get it to look right in the layout for any of the screens (I cleared the cache, rebuilt assets, cleared browser cache etc)

<img width="1682" height="903" alt="Image" src="https://github.com/user-attachments/assets/3d2a1f1d-1949-4df5-a0f0-c8d8791426f4" />
<img width="1645" height="907" alt="Image" src="https://github.com/user-attachments/assets/a7b51c95-cae8-4880-9ba0-184cd000e402" />
<img width="1661" height="870" alt="Image" src="https://github.com/user-attachments/assets/1fbd5b20-24a8-4fed-8503-f128a9a1a132" />
<img width="1664" height="890" alt="Image" src="https://github.com/user-attachments/assets/715179a9-5fb3-414a-910b-a3e2a327569e" />

andersonjeccel committed:
Merge branch '7.x' into lead-space-components

andersonjeccel committed:
fix: apply phpstan, rector and cs-fixer on changed files

andersonjeccel committed:
fix: apply language standards to new files in lead-space-components

- Add final to ColorsExtension, EmailDefaultsHelper, DeleteOrphanFormResultsTableCommand, DeleteOrphanSubmissionRecordsFromFormResultsTableCommand
- Replace hardcoded string in Lead/List/introduction.html.twig with translation key mautic.lead.list.block.unified.heading
- Replace hardcoded violation message in PageHitValidator with translation key mautic.page.hit.validation.missing_fields
- Add mautic.page.hit.validation.missing_fields to PageBundle validators.ini
